msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.props

```diff
@@ -74,6 +76,62 @@ Copyright (C) 2013-2014 Xamarin. All rights reserved.
 		<IsMacEnabled>true</IsMacEnabled>
 	</PropertyGroup>
 
+	<PropertyGroup Condition="'$(XamarinMacFrameworkRoot)' == ''">
+		<XamarinMacFrameworkRoot>/Library/Frameworks/Xamarin.Mac.framework/Versions/Current</XamarinMacFrameworkRoot>
+	</PropertyGroup>
+
+	<Choose>
+		<When Condition=" '$(TargetFrameworkIdentifier)' == 'Xamarin.Mac'">
+			<PropertyGroup>
+				<TargetFrameworkName>Modern</TargetFrameworkName>
+				<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/Xamarin.Mac</MacBclPath>
+			</PropertyGroup>
+		</When>
+		<When Condition=" '$(TargetFrameworkIdentifier)' != 'Xamarin.Mac' And '$(UseXamMacFullFramework)' == 'true'">
+			<PropertyGroup>
+				<TargetFrameworkName>Full</TargetFrameworkName>
+				<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
+			</PropertyGroup>
+		</When>
+		<When Condition=" '$(ForceNonSystemTargetFramework)' == 'true'">
+			<Choose>
+				<When Condition=" '$(TargetFrameworkVersion)' == '2.0' Or '$(TargetFrameworkVersion)' == ''">
+					<PropertyGroup>
+						<TargetFrameworkName>Modern</TargetFrameworkName>
+						<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/Xamarin.Mac</MacBclPath>
+						<TargetFrameworkVersion>v2.0</TargetFrameworkVersion>
+						<TargetFrameworkIdentifier>Xamarin.Mac</TargetFrameworkIdentifier>
+					</PropertyGroup>
+				</When>
+				<Otherwise>
+					<PropertyGroup>
+						<TargetFrameworkName>Full</TargetFrameworkName>
+						<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
+						<UseXamMacFullFramework>true</UseXamMacFullFramework>
+						<AssemblySearchPaths>$(MacBclPath);$(MacBclPath)/Facades</AssemblySearchPaths>
+					</PropertyGroup>
+				</Otherwise>
+			</Choose>
+		</When>
+		<Otherwise>
+			<PropertyGroup>
+				<TargetFrameworkName>System</TargetFrameworkName>
+				<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
+			</PropertyGroup>
+		</Otherwise>
+	</Choose>
+
+	<PropertyGroup Condition="'$(TargetFrameworkName)' == 'Full'">
+		<AssemblySearchPaths>$(XamarinMacFrameworkRoot)/lib/reference/full;$(XamarinMacFrameworkRoot)/lib/mono;$(AssemblySearchPaths)</AssemblySearchPaths>
```

Comment:
Does this work when building from VS2017?

Reply:
I don't have a work VS2017 machine. I guess I could setup everything in a VM, but that'd get messy. The AssemblySearchPaths thing is a hacky i'm hoping to figure out and get rid of.

Reply:
This line was copied directly from msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets (below).